File: src/backend/internal/entity/training.go
package entity

import "github.com/google/uuid"

type TrainingTypeID int

const (
	Aerobic TrainingTypeID = iota
	Anaerobic
	Flexibility
	Strength
)

var TrainingType map[TrainingTypeID]string = map[TrainingTypeID]string{
	Aerobic:     "aerobic",
	Anaerobic:   "anaerobic",
	Flexibility: "flexibility",
	Strength:    "strength",
}

type Training struct {
	ID           uuid.UUID
	Title        string
	Description  string
	TrainingType string
	TrainerID    uuid.UUID
}

func (t *Training) Validate() bool {
	found := false
	for id := range TrainingType {
		if value := TrainingType[id]; value == t.TrainingType {
			found = true
		} 
	}

	if !found {
		return false
	}

	return t.Title != "" && t.Description != ""
}

File: src/frontend/app/src/components/Training/Training.js
import React, { useState, useEffect } from 'react';
import axios from 'axios';
import { useParams, Navigate, useNavigate } from 'react-router-dom';
import './Training.css';

const types = {
  'aerobic': 'аэробная',
  'anaerobic': 'анаэробная', 
  'flexibility': 'гибкость',
  'strength': 'силовая'
}

const Training = ({ isLoggedIn }) => {
  const { trainer_id } = useParams();
  const [trainings, setTrainings] = useState([]);
  const [loading, setLoading] = useState(true);
  const navigate = useNavigate();

  useEffect(() => {
    axios.get(`http://localhost:8080/api/v1/training/${trainer_id}`)
      .then(response => {
        setTrainings(response.data.trainings);
        setLoading(false);
      })
      .catch(error => {
        console.error('Error fetching trainings:', error);
        setLoading(false);
      });
  }, [trainer_id]);

  const toSchedule = (id) => {
    return () => {
      navigate(`/schedule/${id}`)
    } 
  }

  return (
    <div className="training-page">
      <h1>Тренировки</h1>
      {loading ? (
        <p>Загрузка...</p>
      ) : (
        <ul className="training-list">
          {trainings.map(training => (
            <li key={training.ID} className="training-item">
              <h3>{training.Title}</h3>
              <p>Описание: {training.Description}</p>
              <p>Тип: {types[training.TrainingType]}</p>
              {isLoggedIn && (
                <button onClick={toSchedule(training.ID)}>
                  Записаться
                </button>
              )}
            </li>
          ))}
        </ul>
      )}
    </div>
  );
};

export default Training;
